pkg/cmd/admin/router/router.go

```diff
@@ -552,24 +552,46 @@ func RunCmdRouter(f *clientcmd.Factory, cmd *cobra.Command, out, errout io.Write
 	var clusterIP string
 
 	output := cfg.Action.ShouldPrint()
-	generate := output
+	generateSvc := output
 	service, err := kClient.Services(namespace).Get(name)
 	if err != nil {
-		if !generate {
+		if !generateSvc {
 			if !errors.IsNotFound(err) {
 				return fmt.Errorf("can't check for existing router %q: %v", name, err)
 			}
 			if !output && cfg.Action.DryRun {
 				return fmt.Errorf("Router %q service does not exist", name)
 			}
-			generate = true
+			generateSvc = true
 		}
 	} else {
 		clusterIP = service.Spec.ClusterIP
+		fmt.Fprintf(out, "Router %q service exists\n", name)
 	}
 
-	if !generate {
-		fmt.Fprintf(out, "Router %q service exists\n", name)
+	certName := fmt.Sprintf("%s-certs", cfg.Name)
+	generateSecret := output
+	_, err = kClient.Secrets(namespace).Get(certName)
```

Comment:
@liggitt The point of this change is to bail out of the command if it can't be completed. We already check to see if the service already exists and bail if it does. This just checks to see if the secret already exists and bails if it does. 

We either create if from a pem file or use the side effect of a service annotation to create it. If the secret already exists we will just use it, not replace it.